### PR TITLE
Update JTIServer build instructions

### PR DESCRIPTION
**Pull request by dmitry-ten**
- Update build instructions to reflect that JITServer build
process is now nearly identical to regular OpenJ9
- Remove JITServer-related Dockerfiles. They don't bring anything
new compared to regular OpenJ9 Dockerfiles anymore.

Closes: #13317

### CODE REVIEW

**Comment by dmitry-ten** [comment]
@mpirvu ready for review.

File: doc/compiler/jitserver/Build.md [hunk 1]
```diff
@@ -22,159 +22,19 @@ SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-excepti
...
-
-cd "$JIT_SRCBASE"/runtime/compiler
-make -f compiler.mk -C "$JIT_SRCBASE"/compiler -j$(nproc) J9SRC="$JAVA_HOME"/jre/lib/amd64/compressedrefs/ JIT_SRCBASE=.. BUILD_CONFIG=debug J9_VERSION=29 JIT_OBJBASE=../objs
-cp "$JIT_SRCBASE"/compiler/../objs/libj9jit29.so "$J9SRC"
-```
-
-## DOCKER
-
-Follow the [Docker.md](Docker.md) section `build/Dockerfile` for building JITServer in Docker.
+Same instructions as [here](../../build-instructions/Build_Compiler_Only.md),
```
Comment (mpirvu) [inline doc/compiler/jitserver/Build.md:39]: I would add the work `but` before the comma: 'Same instructions as here, but to build with JITServer...`
We may also want to be more specific on what `J9VM_OPT_JITSERVER` needs to be set to.

**Review by mpirvu — COMMENTED** [review]
LGTM. I only have one small comment.

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
This PR only updates documentation so it is safe to deliver and requires no testing.

**Pull request merged**